camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java

```diff
@@ -93,92 +93,140 @@ private static boolean validateParams(OperationParams params) {
+                                ResourceAccessor.open(params.query(), exchange),
+                                outputFilePath);
+                    }
+                    else {
+                        resultFilesPaths = templateExecutor.executeMappingParametric(readers,
+                                ResourceAccessor.open(params.template(), exchange),
+                                ResourceAccessor.open(params.query(), exchange),
+                                outputFilePath);
+                    }
                     exchange.getMessage().setBody(resultFilesPaths, List.class);
+
                 } else {
-                    Path resultFilePath =
-                            templateExecutor.executeMapping(reader, ResourceAccessor.open(params.template(), exchange), outputFilePath, templateMap, formatter, usedTemplateFunctions);
+                    Path resultFilePath;
+                    if (reader != null) {
+                        resultFilePath = templateExecutor.executeMapping(Map.of("reader", reader), ResourceAccessor.open(params.template(), exchange), outputFilePath);
+                    }
+                    else {
+                        resultFilePath = templateExecutor.executeMapping(readers, ResourceAccessor.open(params.template(), exchange), outputFilePath);
+                    }
+
                     exchange.getMessage().setBody(resultFilePath, String.class);
                 }
             }
             // if filename is not specified then the result of applying template is stored in the exchange body as string
             else {
                 if (params.query() != null) {
-                    Map<String,String> result = templateExecutor.executeMappingParametric(reader, ResourceAccessor.open(params.template(), exchange), ResourceAccessor.open(params.query(), exchange), templateMap, formatter, usedTemplateFunctions);
+                    Map<String,String> result;
+                    if (reader != null) {
+                        result = templateExecutor.executeMappingParametric(Map.of("reader", reader),
+                                ResourceAccessor.open(params.template(), exchange),
+                                ResourceAccessor.open(params.query(), exchange));
+                    }
+                    else {
+                        result = templateExecutor.executeMappingParametric(readers,
+                                ResourceAccessor.open(params.template(), exchange),
+                                ResourceAccessor.open(params.query(), exchange));
+                    }
                     exchange.getMessage().setBody(result, Map.class);
 
                 } else {
-                    String result = templateExecutor.executeMapping(reader, ResourceAccessor.open(params.template(), exchange), templateMap, formatter, usedTemplateFunctions);
+                    String result;
+                    if (reader != null) {
```

Comment:
Ideally, define `readers` map at the beginning and then perform calls without checking `reader ` every time